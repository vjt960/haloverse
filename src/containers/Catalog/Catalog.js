import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React from 'react';
import { Card } from '../../components/Card/Card';
import './Catalog.scss';

export const Catalog = props => {
  const enemyCards = props.enemies.map(enemy => {
    return <Card key={`${enemy.name}-${Date.now()}`} {...enemy} />;
  });

  return <section className="catalog">{enemyCards}</section>;
};

export const mapStateToProps = state => ({
  enemies: state.enemies
});

Catalog.propTypes = {
  enemies: PropTypes.array
};

export default connect(mapStateToProps)(Catalog);
