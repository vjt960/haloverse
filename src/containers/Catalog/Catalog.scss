.catalog {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  height: 40vh;
  grid-gap: 10px;
  padding: 20px 20px 0 20px;
  background-color: hsla(51, 100%, 20%, .2);

  // display: grid;
  // grid-template-columns: repeat(3, 1fr);
  // grid-gap: 10px;
  // padding: 20px 20px 0 20px;
  // background-color: hsla(51, 100%, 20%, .2);
  // color: white;
  // height: 50vh;
  // border: 1px solid black;
  // overflow-y: scroll;
}