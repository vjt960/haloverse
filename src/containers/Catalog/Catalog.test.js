import React from 'react';
import { shallow } from 'enzyme';
import { Catalog, mapStateToProps } from './Catalog';

describe('Catalog', () => {
  it('should match snapshot', () => {
    const props = { enemies: [1, 2, 3] };
    const wrapper = shallow(<Catalog {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return array of enemies from state', () => {
      const mockState = { enemies: [1, 2, 3, 4] };
      const expected = { enemies: [1, 2, 3, 4] };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});
