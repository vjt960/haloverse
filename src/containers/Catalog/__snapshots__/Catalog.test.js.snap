// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog should match snapshot 1`] = `ShallowWrapper {}`;
